### PR TITLE
[Refactor][Dataset] LJSPEECH implementation

### PR DESCRIPTION
**Pull request by krishnakalyan3**
See also #910

### CODE REVIEW

**Review by mthrok — COMMENTED** [review]
@krishnakalyan3 

Thanks for working on this. The update looks good. Made some comments.

File: torchaudio/datasets/ljspeech.py [hunk 1]
```diff
@@ -70,14 +53,30 @@ def __init__(self,
...
                     download_url(url, root, hash_value=checksum)
                 extract_archive(archive)
 
         with open(self._metadata_path, "r", newline='') as metadata:
             walker = csv.reader(metadata, delimiter="|", quoting=csv.QUOTE_NONE)
             self._walker = list(walker)
 
+    def _load_item(self, line: List[str], path: str) -> Tuple[Tensor, int, str, str]:
+        assert len(line) == 3
+        fileid, transcript, normalized_transcript = line
```
[the remark this answers is not in the document]
Comment (krishnakalyan3) [reply torchaudio/datasets/ljspeech.py]: I also wanted to know your thoughts on moving from `os.path.join` to  `Path` objects.

[the remark this answers is not in the document]
Comment (mthrok) [reply torchaudio/datasets/ljspeech.py:74]: I think using `Path` as much as possible gives better readability and simpler in some case (like using `glob`).

[the remark this answers is not in the document]
Comment (mthrok) [reply torchaudio/datasets/ljspeech.py:74]: (I would also rename `self._walker` to something like `self._flist` as the variable name `walker` makes no sense.)

**Comment by krishnakalyan3** [comment]
@mthrok is there a way to retrigger the CI test?.

**Comment by mthrok** [comment]
Hi @krishnakalyan3 

Sorry I missed your message. I think can login to CCI with your GitHub account and do "Rerun Workflow form ..."

Screen Shot 2021-01-22 at 18 44 43

**Comment by mthrok** [comment]
@krishnakalyan3 

This looks good. Is it ready to merge?

**Comment by krishnakalyan3** [comment]
@mthrok yes

**Comment by krishnakalyan3** [comment]
should I create a PR for other datasets?. 

My request to join the PyTorch slack channel is still pending. Is there someone I can send an email to?

**Comment by mthrok** [comment]
> should I create a PR for other datasets?.
> 
> My request to join the PyTorch slack channel is still pending. Is there someone I can send an email to?

I pinged the team, so hopefully your request will be processed soon. If that's not the case, please email moto@fb.com and I will follow up.

**Comment by mthrok** [comment]
Thanks!

**Comment by mthrok** [comment]
> should I create a PR for other datasets?.

Yes, go ahead.

**Comment by mthrok** [comment]
@krishnakalyan3 Can you send me (moto@fb.com) the email you are trying to use for joining PyTorch slack channel?

**Pull request merged**